h/test/session_test.py

```diff
@@ -1,47 +1,90 @@
-import unittest
-
 import pytest
-from mock import Mock
-from mock import patch
+import mock
 
-from h.session import model
+from h import session
 
 
-class FakeGroup():
+class FakeGroup(object):
     def __init__(self, pubid, name):
         self.pubid = pubid
         self.name = name
         self.slug = pubid
 
 
-@patch('h.models.User')
-def test_sorts_groups(User):
-    fake_user = Mock()
+@mock.patch('h.models.User', autospec=True)
+def test_model_sorts_groups(User):
+    fake_user = mock.Mock()
     fake_user.groups = [
         FakeGroup('c', 'Group A'),
         FakeGroup('b', 'Group B'),
         FakeGroup('a', 'Group B'),
     ]
-    request = Mock(authenticated_user=fake_user)
-    session_model = model(request)
+    request = mock.Mock(authenticated_user=fake_user)
+    session_model = session.model(request)
 
     ids = [group['id'] for group in session_model['groups']]
+
     assert ids == ['__world__', 'c', 'a', 'b']
 
 
-@patch('h.session.features')
-def test_includes_features(features, fake_user):
+@mock.patch('h.session.features', autospec=True)
+def test_model_includes_features(features, fake_user):
     feature_dict = {
         'feature_one': True,
         'feature_two': False,
     }
-    features.all = Mock(return_value=feature_dict)
-    request = Mock(authenticated_user=fake_user)
-    assert model(request)['features'] == feature_dict
+    features.all = mock.Mock(return_value=feature_dict)
```

Comment:
I find this very verbose. We essentially have a matrix of conditions and expected outputs:

```
tutorial flag enabled? | tutorial dismissed? | expect show_tutorial?
N | N | N
N | Y | N
Y | N | Y
Y | Y | N
```
Is there a facility that we could use to define data driven tests with a truth table like the above, do we have any existing instances of this?